Update mod-1v1-arena

milestorme opened the pull request:
few fix's
Removing double define , we don't just rename ScriptPch.h to ScriptMgr.h we delete if its already defined, , convert to English and fix other English.

FrancescoBorzi commented on Sql/1v1_Battlemaster.sql:
```diff
@@ -1,2 +1,4 @@
+DELETE FROM `creature_template` WHERE `entry` = 999990;
+DELETE FROM `creature_template` WHERE `entry` = 999991;
```
those two lines can be combined:

http://www.azerothcore.org/wiki/Dealing-with-SQL-files#write-compact-code

everything else looks good

milestorme replied:
done :)

milestorme commented:
`[ 42%] Building CXX object src/server/scripts/CMakeFiles/scripts.dir/__/__/__/modules/mod-1v1-arena/src/npc_arena1v1.cpp.o
/var/michael/corerocket/azerothcore-wotlk/modules/mod-1v1-arena/src/npc_arena1v1.cpp:256:89: warning:
      flag ' ' results in undefined behavior with 'u' conversion specifier [-Wformat]
  ...ChatHandler(player->GetSession()).PSendSysMessage("You have to be level% u + to create a 1v...
                                                                            ~^~
/var/michael/corerocket/azerothcore-wotlk/modules/mod-1v1-arena/src/npc_arena1v1.cpp:306:67: warning:
      format string is not a string literal (potentially insecure) [-Wformat-security]
                ChatHandler(player->GetSession()).PSendSysMessage(s.str().c_str());
                                                                  ^~~~~~~~~~~~~~~
2 warnings generated.

still 2 warnings I couldn't fix yet`

milestorme commented:
1 warning left 

`/var/michael/corerocket/azerothcore-wotlk/modules/mod-1v1-arena/src/npc_arena1v1.cpp:306:67: warning:
format string is not a string literal (potentially insecure) [-Wformat-security]
ChatHandler(player->GetSession()).PSendSysMessage(s.str().c_str());`

milestorme commented:
its not reading any of the conf

`-> Not found option 'CONFIG_ARENA_1V1_ENABLE'. The default value is used (Yes)
-> Not found option 'CONFIG_ARENA_1V1_COSTS'. The default value is used (400000)
-> Not found option 'CONFIG_ARENA_1V1_MIN_LEVEL'. The default value is used (70)
-> Not found option 'CONFIG_ARENA_1V1_MIN_LEVEL'. The default value is used (70)
-> Not found option 'CONFIG_ARENA_1V1_ENABLE'. The default value is used (Yes)
-> Not found option 'CONFIG_ARENA_1V1_COSTS'. The default value is used (400000)
-> Not found option 'CONFIG_ARENA_1V1_BLOCK_FORBIDDEN_TALENTS'. The default value is used (Yes)
-> Not found option 'CONFIG_ARENA_1V1_MIN_LEVEL'. The default value is used (70)`

milestorme commented:
WIP. Don’t merge anything yet , I’ll get conf working first and add the missing gitattributes and such

milestorme commented:
added .gitattributes ,gitignore .travis.yml fixed the sql updated cmake, added correct conf loading, still wont read from the conf file. but is tested and works , will be more stable with the few minor fix's that where causing warnings

BarbzYHOOL commented:
hmm not reading the conf on windows/linux? both?

milestorme commented:
On both , it prints to console that silly warning and uses the default values in the cpp

BarbzYHOOL commented:
Maybe ask Winfidonarleyan (on my behalf lol)

milestorme commented:
this is complete and ready for review to merge

BarbzYHOOL commented on Sql/1v1_Battlemaster.sql:
```diff
@@ -1,2 +1,3 @@
-INSERT INTO `creature_template` (`entry`, `difficulty_entry_1`, `difficulty_entry_2`, `difficulty_entry_3`, `KillCredit1`, `KillCredit2`, `modelid1`, `modelid2`, `modelid3`, `modelid4`, `name`, `subname`, `IconName`, `gossip_menu_id`, `minlevel`, `maxlevel`, `exp`, `faction`, `npcflag`, `speed_walk`, `speed_run`, `scale`, `rank`, `mindmg`, `maxdmg`, `dmgschool`, `attackpower`, `DamageModifier`, ` ...
+DELETE FROM `creature_template` WHERE `entry` IN (999991);
```
??? @milestorme why?
I don't see why but you can maybe explain it.

milestorme replied:
Ok according to xdev and my testing this npc does nothing , and can be removed only 1 npc is needed for this script , hence not needed

BarbzYHOOL replied:
What if someone has created a creature_template ID with the same ID, here it just deletes the row but doesn't add anything on top of it, very confusing

BarbzYHOOL replied:
I think it's unsafe @milestorme

milestorme replied:
Your referencing my original typo I delete 99991 and the sql insert adds 99991 not 99990 that was an early typo I fixed later as we had 2 entries before check my latest pr1 I changed that half way through

milestorme replied:
You want I can remove the delete function just incase someone has used this custom id for something else

milestorme replied:
I just removed it for safety

BarbzYHOOL replied:
In fact, my first comment was erroneous because (I'm ill and) I didn't check properly the initial SQL. But after a 2 sec analysis, I think it's still better to not delete, because if someone has used that ID, then it will print an error and the user will just need to change it and that's fine considering the SQL file is only 2 lines long.

What do you think?

milestorme replied:
Yeah I agree , I removed it

BarbzYHOOL commented on src/cmake/after_ws_install.cmake:
```diff
@@ -13,3 +13,4 @@ if( WIN32 )
 endif()
 
 install(FILES "${CMAKE_1V1ARENA_DIR}/conf/1v1arena.conf.dist" DESTINATION ${CONF_DIR})
+
```
If you added a space on purpose, it's not entirely needed if it has a carriage return at the end (\r different from a new line \n)

milestorme replied:
My mistake , copy paste typo

BarbzYHOOL commented on src/npc_arena1v1.h:
```diff
@@ -60,7 +60,7 @@ static bool Arena1v1CheckTalents(Player* player)
     if (count >= 36)
     {
-        ChatHandler(player->GetSession()).SendSysMessage("Du kannst nicht beitreten, da du zuviele Talentpunkte in einem verbotenen Baum hast. (Heal / Tank)");
+        ChatHandler(player->GetSession()).SendSysMessage("You can not join because you have too many talent points in a forbidden tree. (Heal / Tank)");
```
Too many talent points in a forbidden tree ? if it's forbidden, it's 0 talent allowed in that tree no ? if so then the phrase is redundant :P

milestorme replied:
To tell you truth , I’m not sure why this is set like this either , or how I would set it , I just converted the German to English on this line , appreciate and input others have here in this.

milestorme replied:
ok on further research this is a conf setting regarding extra points "Description: If true, healers can't join 1v1 arena, if they invested more than 35 talent points in a healing-talent tree." 

so this is possible adjusted +1 to compensate for something or this was a custom change to add 36 not 35 as originally intended

BarbzYHOOL commented:
Very good improvement on the original module, thank you! I left few comments

milestorme commented:
cleaned up spaces , added missing in game announcement is already in arena team, re did the announcer as no override was causing no in game module announce whether set to 0 or 1 , hopefully I haven't missed anything still , please test in game to confirm I have and all seems to be working well on Debian 9
